### ISSUE TITLE
System theme not available for new profile in Windows

### ISSUE THREAD

**Issue by Wonderer0** [MrAlex94/Waterfox#2416]
### What happened?

If you create a new profile in 4.0.2.1 and look in `about:addons` the built-in System theme is no longer available.  If you use an old profile (created before G4) it is still available, but gives different colours in the tab bar etc, as reported in issue #2369.  

I am reporting this as a "bug" in the hope that it's not been deliberately removed.  If it has could you restore it please, not only to make WF look similar to other applications, also to enable userChrome.css based themes work as expected.

This is very similar to issue #2393, but I am creating a new issue as that one implies it's only a problem in GTK.

The System theme is present in Firefox (94.0.1) as the default theme for a new profile, and colours are as expected.

### Reproducible?

- [X] I have tested this on Firefox.

### Version

G4.𝑋 (Future)

### What platform are you seeing the problem on?

Windows

**Comment by Wonderer0** [comment]
### Workaround

In case anybody else runs into this problem a workaround is given below.  Thanks to @Kilzzz for his/her workaround, which made me realise it's possible to disable all themes.

#### Summary

The workaround is to install an add-on theme, then just delete it's `.xpi` file from `<profile>\extensions`.  Waterfox can then no longer use this theme, but it leaves all the other themes disabled rather than reverting to the default one.  Having all themes disabled seems to have the same effect as using the System theme. 

WF leaving all themes disabled is perhaps a bug itself, but let's hope it's not fixed until the System Theme is restored, or it's possible to disable all themes in some other way through the GUI.

#### Steps to follow:

Many themes have filenames that are auto-generated GUIDs, which makes it difficult to tell which is which.  So it's easier to create your own with a known name, e.g. `empty-theme@nowhere` (must contain "@").

1. Create a `manifest.json` file:
```
{
  "manifest_version": 2,
  "version": "0.1",
  "name": "Empty Theme",
  "theme": {},
   "browser_specific_settings": { "gecko": { "id": "empty-theme@nowhere" }  }
} 
```

2. Compress this into a zip archive (e.g. from Windows Explorer using 7-zip's `Add to "manifest.zip"`).
3. Change the extension to `.xpi` (e.g. `manifest.xpi`).
4. Double click to install it.  WF should now be using Empty Theme.
5. Shutdown WF.
6. Delete `<profile>\extensions\empty-theme@nowhere.xpi`.

When you restart WF the Empty Theme will have vanished from `about:addons`, but all the other themes will also still be disabled.  Fortunately this persists through multiple restarts.

**Comment by Kilzzz** [comment]
Sadly the work around no longer works in 4.0.8. Instead it leaves a nasty on the eyes white theme. 

[This extension](https://addons.mozilla.org/en-US/firefox/addon/firefox-color/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search) will work to change the theme. Sadly I was not able to save and apply the theme for the future. But its easy to use and change the theme.

**Comment by Wonderer0** [comment]
@Kilzzz Thanks for the link, but unfortunately I couldn't see how to change only a few parts, such as the just selected tab's colour, using [Firefox Color](https://addons.mozilla.org/en-US/firefox/addon/firefox-color/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search).  You have to choose a preset theme and then change every part of that to make it roughly like the system theme. 

If I understand correctly, the way it used to work was that Waterfox's System Theme acted like minimal layer that allowed the operating system's current theme to show thorough, and individual parts (like selected tab) could be tweaked as desired in `userChrome.css` without changing anything else.  It's a great pity that the System Theme has been removed, even though it's still available in Firefox..

**Comment by Kilzzz** [comment]
The extension is pretty easy. Once it starts up click on the custom colors tab. Clicking on anything, like toolbar color brings up a color selector on the right. You can put in rgb or hex colors and it applies it to waterfox. The advanced colors has some nice options as well. Once you save the theme you can export it, but it needs to be signed by mozilla to be installed from a file. [Getting it signed is real easy here. ](https://addons.mozilla.org/en-US/developers/themes) . Getting a theme signed may not be helpful to most people, but I create spins of Manjaro linux and a firefox/waterfox theme that matches the system theme is something I am going to be offering.

**Comment by Wonderer0** [comment]
I looked at the "Advanced colors" in the hope that they would give a way of choosing [System Colors](https://www.w3.org/wiki/CSS/Properties/color/keywords#System_Colors) (e.g. `Menu`, `Highlight`) instead of just RGB values that correspond to those colours.  The advantage of using System Colors is that they change automatically if you change your desktop theme, so WF always looks similar to other apps.

However I have discovered that if you create a theme in a `manifest.json` file (described [here](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/theme)) the System Colors  do work as expected.  As I only wanted to change a just few colours this was the easiest way to do it.  Creating a theme using `userChrome.css` gives more fine grain control (e.g. Unselected tabs different from `frame`), but this way is good enough.

N.B. As my theme is is just for my personal use in WF there's no need to get it signed.  I just added a line `"browser_specific_settings": { "gecko": { "id": "my-system-theme@nowhere" }  }`
and installed it as an extension as described above.
